### PR TITLE
[Galley] Restructure runtime package to support multiple states.

### PR DESCRIPTION
**Pull request by nmittler**
This is a follow-on to #11162 that moves the runtime state as well as
 its previously package-private dependencies into their
 own packages. This allows new "states" to exist in separate packages
 under runtime.

This is supporting work for #11293, which is addressing both #10497 and #10589

### CODE REVIEW

File: galley/pkg/runtime/states/defaultstate/state.go [hunk 1]
```diff
@@ -73,21 +72,22 @@ type resourceTypeState struct {
...
-func newState(name string, schema *resource.Schema, cfg *Config, strategy *publishingStrategy,
-	distributor Distributor) *State {
+// New creates a new State
+func New(name string, domainSuffix string, schema *resource.Schema, strategy *publish.Strategy,
+	distributor publish.Distributor) *State {
 
 	now := time.Now()
 	s := &State{
 		name:             name,
+		domainSuffix:     domainSuffix,
```
Comment (ozevren) [inline galley/pkg/runtime/states/defaultstate/state.go]: Are you replacing config with domainSuffix due to cyclic dependency problems?

File: galley/pkg/runtime/state.go [hunk 2]
```diff
@@ -73,21 +72,21 @@ type resourceTypeState struct {
 	versions map[resource.FullName]resource.Version
 }
 
-func newState(name string, schema *resource.Schema, cfg *Config, strategy *publishingStrategy,
-	distributor Distributor) *State {
+func newState(name string, domainSuffix string, schema *resource.Schema, strategy *publish.Strategy,
```
Comment (ozevren) [inline galley/pkg/runtime/state.go]: Nit: You can reintroduce the config here.

File: galley/pkg/runtime/state.go [hunk 3]
```diff
@@ -27,23 +27,22 @@ import (
...
 	"istio.io/istio/galley/pkg/runtime/conversions"
-	"istio.io/istio/galley/pkg/runtime/processing"
+	"istio.io/istio/galley/pkg/runtime/log"
+	"istio.io/istio/galley/pkg/runtime/monitoring"
+	"istio.io/istio/galley/pkg/runtime/publish"
 	"istio.io/istio/galley/pkg/runtime/resource"
 	"istio.io/istio/pkg/mcp/snapshot"
 )
 
-var _ processing.Handler = &State{}
```
Comment (ozevren) [inline galley/pkg/runtime/state.go]: This still applies, right?

Comment (nmittler) [reply galley/pkg/runtime/state.go]: done

File: galley/pkg/runtime/states/defaultstate/state.go (hunk 1 above)
Comment (nmittler) [reply galley/pkg/runtime/states/defaultstate/state.go]: no, just noticed only a part was needed ... reverted.

**Pull request merged**